multi/stm32l4-multi/flash.c

```diff
@@ -160,38 +187,53 @@ static int _program_erasePage(uint32_t addr)
 }
 
 
-static int _program_writePage(uint32_t offset)
+static int _program_dblWords(volatile uint32_t *addr, const char *buff, size_t size)
 {
 	int pos = 0;
-	volatile uint32_t *ptr = (void *)offset;
-	uint32_t t[2];
-
-	if (offset & (FLASH_PAGE_SIZE - 1))
-		return -1;
-
-	if (_flash_wait() != 0)
-		return -1;
-
-	_flash_clearFlags();
+	uint32_t value[2];
+	int ret;
 
-	for (pos = 0; pos < FLASH_PAGE_SIZE; pos += 2 * sizeof(uint32_t)) {
+	for (pos = 0; pos < size; pos += 2 * sizeof(uint32_t)) {
 		*(flash_common.flash + flash_cr) |= 1;
-		memcpy(t, flash_common.page + pos, 2 * sizeof(uint32_t));
-		*(ptr++) = t[0];
+		memcpy(value, buff + pos, 2 * sizeof(uint32_t));
+		*(addr++) = value[0];
 		dataBarier();
-		*(ptr++) = t[1];
+		*(addr++) = value[1];
 		dataBarier();
 
-		if (_flash_wait() != 0) {
-			*(flash_common.flash + flash_cr) &= ~1;
-			return -1;
+		ret = _flash_wait();
```

Comment:
ret is ignored anyway
```suggestion
		if (_flash_wait() != 0) {
```

Reply:
Done